charts/rancher-istio/rancher-istio/100.0.0/charts/kiali/templates/_helpers.tpl

```diff
@@ -50,8 +50,20 @@ app.kubernetes.io/part-of: "kiali"
 Selector labels
 */}}
 {{- define "kiali-server.selectorLabels" -}}
+{{- $releaseName := .Release.Name -}}
+{{- $fullName := include "kiali-server.fullname" . -}}
 app.kubernetes.io/name: kiali
-app.kubernetes.io/instance: {{ include "kiali-server.fullname" . }}
+{{- if .Release.IsUpgrade }}
+    {{- $labels := (lookup "apps/v1" "Deployment" .Release.Namespace $fullName).metadata.labels }}
```

Comment:
Since these labels are contained in a helper function, can we assume that the deployment is the only resource we need to check to get a preserved `$instanceValue`? Or do we need to do multiple lookups?

Reply:
possibly should look at the service instead, based on offline discussion